### DISCUSSION TITLE
Get "Planned Charge from Grid" data from a Service like Tibber or entso-e

### DISCUSSION THREAD

**Discussion by CommanderROR9** [wlcrs/huawei_solar#255] [Ideas]
Hi,

I am currently working out the best way to use cheap, green electricity by charging my Luna2000 battery when energy is cheap (of course always keeping in mind how much PV production there will he so the battery won't be full ready) and am doing so using the "forcible charge to SoC" entity. The question is...is there a more elegant solution? Maybe writing a daily Charging Schedule by getting data from the entso-e or Tibber integrations? I have no idea how to even start doing this or if it's even worth it, but wanted to put the idea out there...

**Comment by Rhysgh** [comment]
I'm doing the same thing, I have a force charge option working, but I need to lean how to automate a TOU update.
